start.sh

```diff
@@ -2,14 +2,52 @@
 
 set -uo pipefail
 
+# Directory for self-signed certificates
+CERT_DIR="Docker/certs"
+CERT_FILE="$CERT_DIR/server.crt"
+KEY_FILE="$CERT_DIR/server.key"
+
+# Create the certificate directory if it doesn't exist
+mkdir -p $CERT_DIR
+
+# Generate self-signed certificates if they don't exist
+if [[ ! -f "$CERT_FILE" || ! -f "$KEY_FILE" ]]; then
+    echo "Generating self-signed certificates"
+    openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout $KEY_FILE -out $CERT_FILE -subj "/CN=localhost" -addext "subjectAltName = DNS:localhost,IP:127.0.0.1"
+
+    # Attempt to add the certificate to the trusted store
+    if [[ "$(uname)" == "Linux" ]]; then
+        echo "Adding certificate to the trusted store"
+        cp $CERT_FILE /usr/local/share/ca-certificates/
+        update-ca-certificates
+    elif [[ "$(uname)" == "Darwin" ]]; then
+        echo "Adding certificate to the trusted store (Mac)"
+        sudo security add-trusted-cert -d -r trustRoot -k /Library/Keychains/System.keychain $CERT_FILE
+    else
+        echo "Unsupported platform for adding certificates to the trusted store"
+    fi
+else
+    echo "Self-signed certificates already exist"
+fi
+
+# Bring down any existing services
 docker compose down
 
-docker build .
+# Build the Docker image
+docker compose build
 
+# Start the services
 docker compose up -d
 
+# Wait for the MySQL container to be fully up
+echo "Waiting for MySQL to be ready..."
+until docker exec db mysqladmin ping -h "db" --silent; do
```

Comment:
I'm 99.5% sure it can only be run locally for our testing instance - including a counter is a good idea, I could do that, but when I've been testing it usually loads the db within ~7 tries, which is on par with how it did previously